src/serialization/KiotaSerializer.Serialization.cs

```diff
@@ -0,0 +1,81 @@
+// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+#if NET5_0_OR_GREATER
+using System.Diagnostics.CodeAnalysis;
+#endif
+
+
+namespace Microsoft.Kiota.Abstractions.Serialization;
+
+/// <summary>
+/// Set of helper methods for serialization
+/// </summary>
+public static partial class KiotaSerializer
+{
+    /// <summary>
+    /// Serializes the given object into a string based on the content type.
+    /// </summary>
+    /// <typeparam name="T">Type of the object to serialize</typeparam>
+    /// <param name="contentType">Content type to serialize the object to </param>
+    /// <param name="value">The object to serialize.</param>
+    /// <returns>The serialized representation as a stream.</returns>
+    public static Stream SerializeAsStream<T>(string contentType, T value) where T : IParsable
+    {
+        using var writer = GetSerializationWriter(contentType, value);
+        writer.WriteObjectValue(string.Empty, value);
+        return writer.GetSerializedContent();
+    }
+    /// <summary>
+    /// Serializes the given object into a string based on the content type.
+    /// </summary>
+    /// <typeparam name="T">Type of the object to serialize</typeparam>
+    /// <param name="contentType">Content type to serialize the object to </param>
+    /// <param name="value">The object to serialize.</param>
+    /// <returns>The serialized representation as a string.</returns>
+    public static string SerializeAsString<T>(string contentType, T value) where T : IParsable
+    {
+        using var stream = SerializeAsStream(contentType, value);
+        return GetStringFromStream(stream);
+    }
+    /// <summary>
+    /// Serializes the given object into a string based on the content type.
+    /// </summary>
+    /// <typeparam name="T">Type of the object to serialize</typeparam>
+    /// <param name="contentType">Content type to serialize the object to </param>
+    /// <param name="value">The object to serialize.</param>
+    /// <returns>The serialized representation as a stream.</returns>
+    public static Stream SerializeAsStream<T>(string contentType, IEnumerable<T> value) where T : IParsable
```

Comment:
Out of curiosity, should we have the `SerializeAsStream` create and return the stream? Or should we let the caller pass the stream and we simply write to it here?(This may also allow us to simple have overloads of the `Serialize` method instead of a `SerializeAsStream` and a `SerializeAsString` method similar to )

Reply:
This is a design flaw we'll need to address in v2. Basically the serialization writer infrastructure doesn't allow to pass a stream to write directly to. So if we accepted a stream here, the only thing we'd be doing is copying from the one returned by the serialization writer to this one.

Reply:
Yeah. I see that. The extra copy work may not be worth it now.